p2p/trust/trustmetric.go

```diff
@@ -144,6 +144,7 @@ func (tms *TrustMetricStore) loadFromDB() bool {
 	for key, p := range peers {
 		tm := NewMetricWithConfig(tms.config)
 
+		tm.mtx.Lock()
```

Comment:
Could we perhaps extract the code below into a helper method such as:
```go
func (tms *TrustMetricStore) recalculate(p *peerHistoryJSON) {
   tms.mtx.Lock()
   defer tms.mtx.Unlock()

   // Restore the number of time inteverals we have previously tracked
   ...
```
and then in that loop
```go
for key, p := range peers {
     tm := NewMetricWithConfig(tms.config)
     tm.recalculate(&p)

     // Save the updated peer
     peers[key] = p
```
or someway of already having the pointer to p.

I am making this suggestion to ease with the sparse locking.